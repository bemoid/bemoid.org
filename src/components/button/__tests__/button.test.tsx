import { screen, render, fireEvent } from '@testing-library/react'

import { Button, ButtonFill, ButtonHollow } from '..'

describe('Buttons components', () => {
  it('matches snapshot', () => {
    const { asFragment: fragmentButton } = render(
      <Button href="https://bemoid.org/">Button</Button>
    )
    const { asFragment: fragmentButtonFill } = render(
      <ButtonFill href="https://bemoid.org/">Button Fill</ButtonFill>
    )
    const { asFragment: fragmentButtonHollow } = render(
      <ButtonHollow href="https://bemoid.org/">Button Fill</ButtonHollow>
    )

    expect(fragmentButton()).toMatchSnapshot()
    expect(fragmentButtonFill()).toMatchSnapshot()
    expect(fragmentButtonHollow()).toMatchSnapshot()
  })

  it('calls `onClick` event', () => {
    const handleClick = jest.fn()

    render(
      <Button href="https://bemoid.org/" onClick={handleClick}>
        Button
      </Button>
    )

    const button = screen.getByTestId('button')

    fireEvent.click(button)
    expect(handleClick).toHaveBeenCalled()
  })
})
