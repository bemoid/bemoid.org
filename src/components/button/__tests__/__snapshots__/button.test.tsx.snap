// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons components matches snapshot 1`] = `
<DocumentFragment>
  <a
    class=" css-ybdpjn"
    data-testid="button"
    href="https://bemoid.org/"
  >
    Button
  </a>
</DocumentFragment>
`;

exports[`Buttons components matches snapshot 2`] = `
<DocumentFragment>
  <a
    class="css-f6v8k"
    data-testid="button"
    href="https://bemoid.org/"
  >
    Button Fill
  </a>
</DocumentFragment>
`;

exports[`Buttons components matches snapshot 3`] = `
<DocumentFragment>
  <a
    class="css-xmjzch"
    data-testid="button"
    href="https://bemoid.org/"
  >
    Button Fill
  </a>
</DocumentFragment>
`;
